ui/src/helpers/get-issue-category.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2025 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+export type IssueCategory =
+  | 'Build System'
+  | 'Missing Data'
+  | 'Infrastructure'
+  | 'Other';
+
+/**
+ * Use regular expressions to categorize an issue based on its message.
+ *
+ * @param message The issue message
+ * @returns The issue category
+ */
+export const getIssueCategory = (message: string): IssueCategory => {
+  // These regular expressions are used to categorize the issues
+  // and they will be expanded as new cases are discovered.
+  const infrastructureRegEx: RegExp = new RegExp(
+    'The .* worker failed due to an unexpected error.*' +
+      "|ERROR: Timeout after .* seconds while scanning file '.*'\\."
+  );
+  const missingDataRegEx: RegExp = new RegExp(
+    'IOException: Could not resolve provenance for .*'
+  );
+  const buildSystemRegEx: RegExp = new RegExp(
+    '.* failed to resolve dependencies for .*'
+  );
+
+  switch (true) {
+    case infrastructureRegEx.test(message):
+      return 'Infrastructure';
+    case missingDataRegEx.test(message):
+      return 'Missing Data';
+    case buildSystemRegEx.test(message):
+      return 'Build System';
+    default:
+      return 'Other';
+  }
```

Comment:
As it's not immediately obvious what `switch (true)` does here and there's no need for fall through, I'd replace this with if-elseif-else or ifs with early returns.